### PR TITLE
chore: merge dev to cdkv2 with category api and graphql update

### PR DESCRIPTION
**Pull request by sobolk**
#### Description of changes

This PR:
- merges dev to cdkv2 branch
- upgrades data dependencies
- upgrades graphql to 15.x

image

The graphql is referenced by both cli and data packages. Therefore merge (which brings graphql update in cli) and category api bump goes together in this PR. Otherwise graphql version mismatch causes schema generation failures at runtime.

#### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies
- [ ] [Pull request labels](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#labels) are added

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by jhockett — APPROVED** [review]

File: packages/amplify-util-mock/package.json [hunk 1]
```diff
@@ -29,16 +29,16 @@
     "extract-api": "ts-node ../../scripts/extract-api.ts"
   },
   "dependencies": {
-    "@aws-amplify/amplify-appsync-simulator": "2.9.1",
-    "@aws-amplify/amplify-opensearch-simulator": "1.2.1",
+    "@aws-amplify/amplify-appsync-simulator": "2.9.3",
+    "@aws-amplify/amplify-opensearch-simulator": "1.1.4",
```
Comment (akshbhu) [inline packages/amplify-util-mock/package.json:33]: this should be 1.2.1 I think ?

**Review by akshbhu — COMMENTED** [review]
looks good , just one question

File: packages/amplify-util-mock/package.json (hunk 1 above)
Comment (sobolk) [reply packages/amplify-util-mock/package.json:33]: yep good catch.

Comment (sobolk) [reply packages/amplify-util-mock/package.json:33]: I'm gonna ammend this to last commit.

**Review by akshbhu — APPROVED** [review]

**Review by awsluja — APPROVED** [review]
looks good, what is the motive behind the lerna bump (just maintenance im assuming?); its non-breaking https://github.com/lerna/lerna/releases/tag/v6.0.0
so I was just curious

**Comment by sobolk** [comment]
> looks good, what is the motive behind the lerna bump (just maintenance im assuming?); its non-breaking https://github.com/lerna/lerna/releases/tag/v6.0.0 so I was just curious

This came from dev https://github.com/aws-amplify/amplify-cli/pull/11829 .

**Pull request merged**